### ISSUE TITLE
Templates: location information from @angular/compiler is not always as expected

### ISSUE THREAD

**Issue by drakenfly** [angular-eslint/angular-eslint#292]
I noticed that expressions with additional whitespaces lead to wrong reports of errors:

I created an examplatory test-case for the negated-async pipe which fails, even though it should obviously pass:
```
convertAnnotatedSourceToFailureCase({
  description: 'it should fail if async pipe is negated even with additional whitespace',
  annotatedSource: `
    {{     !(foo | async) }}
           ~~~~~~~~~~~~~~
  `,
  messageId: noNegatedAsync
}),
```
Note: This also affects BoundAttributes, not only Interpolations.

@angular-eslint/template-parser invokes @angular/compiler with preserveWhitespaces set to true:

```
function parseForESLint(code: string, options: { filePath: string }) {
  const angularCompilerResult = parseTemplate(code, options.filePath, {
    preserveWhitespaces: true,
  });
 // ...
}
```

I tried to debug the parser and it seems impossible to find the absolute positions in the actual sourcecode with preserveWhitespaces: true, because the source-property of the AST is trimmed before the compiler decides about the location of the processed source code fragment.

Is there a way to solve this issue? Simply switching it off leads to a lot off issues in other rules (I tried running all test cases in the project).

**Comment by JamesHenry** [comment]
There isn't documentation from the Angular Team for usage of the compiler API, so everything we know/have implemented is based on just digging in and trying to figure stuff out.

I will message the Angular tooling team and request a process for us to report and resolve these issues.

**Comment by JamesHenry** [comment]
From @rafaelss95 on an overlapping issue:

---

As reported in https://github.com/mgechev/codelyzer/issues/859, https://github.com/angular-eslint/angular-eslint/pull/205#discussion_r532132323 and https://github.com/angular-eslint/angular-eslint/pull/269#discussion_r548901096, column numbers are off by one.

Sample:

https://github.com/angular-eslint/angular-eslint/blob/b7ea86c6516d3ef8e07f98cfbd0c453970018b64/packages/eslint-plugin-template/tests/rules/no-call-expression.test.ts#L44-L45

<hr>

I noticed that we have a _hacky_ way to handle this in `no-negated-async`:

https://github.com/angular-eslint/angular-eslint/blob/c8dd14c95f881593d7b6ca9f7ba80b37b7eaf40f/packages/eslint-plugin-template/src/rules/no-negated-async.ts#L40 and https://github.com/angular-eslint/angular-eslint/blob/c8dd14c95f881593d7b6ca9f7ba80b37b7eaf40f/packages/eslint-plugin-template/src/rules/no-negated-async.ts#L57-L58

**Comment by JamesHenry** [comment]
Folding this in, even though it might not be 100% related to the original description.

Truly single line inline-templates have wonky location information right now:

---

✅  **Multi-line:**
![image](https://user-images.githubusercontent.com/900523/97445667-6c83e680-1925-11eb-9f9c-80029c70308b.png)

🙅  **Single-line:**
![image](https://user-images.githubusercontent.com/900523/97445753-845b6a80-1925-11eb-89d0-497fd442111b.png)

**Comment by jerone** [comment]
Not only column is off by one (or sometimes two):

```
test
{{ property }}
  ~~~~~~~~
```

The line index can also be off:

```
test
{{
  ~~~~~~~~
    property
}}
```

**Comment by kyliau** [comment]
I think this might have something to do with CRLF line endings.
@JamesHenry could you please try passing `preserveLindEndings: true` to `parseTemplate` and see if it fixes the offset problem?
Ref https://github.com/angular/angular/blob/66c27ffdfc0e4f0a37a375a453d368c8ef13e326/packages/compiler/src/ml_parser/lexer.ts#L116-L119

**Comment by JamesHenry** [comment]
Thanks @kyliau! At the time you wrote this it hadn't actually been released :)

It is now available on npm though and I have a PR in the works which enables that and preserveWhitespace. With those two we definitely seem to get more predictable location information, thanks a lot!

If you could let us know your/the teams thoughts on HTML comment nodes that would cover off everything for now

Thanks again!

**Comment by JamesHenry** [comment]
Resolved in the newly published `v2.0.0`, see releases or changelog for more information.

**Comment by jerone** [comment]
I upgraded my plugin and can confirm that all **interpolation** locations are correct now. Thanks.

However, the location for **data-binding** is still off.   
Let me explain with the following HTML template:

```html
<div [class]="
    value
">
</div>
```
As seen in the screenshot below, the location of the warning is:

```json
{
	"start": { 
		"line": 2, 
		"column": 9 
	}, 
	"end": { 
		"line": 3, 
		"column": 3 
	}
}
```

```html
<div [class]="
    value
         ~
">
~~
</div>
```
![image](https://user-images.githubusercontent.com/55841/111079390-2016fd00-84fa-11eb-9f7f-617fc44fb74d.png)

While, I expect the error to be under value:

```json
{
	"start": { 
		"line": 2, 
		"column": 4 
	}, 
	"end": { 
		"line": 2, 
		"column": 9 
	}
}
```

```html
<div [class]="
    value
    ~~~~~
">
</div>
```

**Comment by jerone** [comment]
@JamesHenry Should I open an new bug issue with above report?

**Comment by JamesHenry** [comment]
Yes please
